staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -476,7 +476,18 @@ type DeploymentStatus struct {
 	// +optional
 	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty" protobuf:"varint,5,opt,name=unavailableReplicas"`
 
-	// Represents the latest available observations of a deployment's current state.
+	// Represents the latest observations of a deployment's current state (described by the observedGeneration).
+	// Deployments are given the following conditions automatically:
+	// Available is true when the number of available replicas under the deployment's replica sets is
+	// greater than the user's defined minimum availability. The minimum availability is controlled by the
+	// update strategy - the Recreate strategy has a minimum availability of zero (always Available), and
+	// the Rolling strategy defaults to a minimum availability of zero unless replicas is greater than 1,
```

Comment:
> // Absolute number is calculated from percentage by rounding down.
> // Defaults to 25%.
> MaxUnavailable

it defaults to 25% `maxUnavailable` and 25% `maxSurge` so in case replicas is 0-3 the minimum availability will be 100%

Reply:
Coming back to this comment, now I'm confused (which doesn't bode well).  I think you're saying "the fencepost protection for 25% ensures that if `total pods * 25% < 1, your minimum availability is 1"?

Reply:
I see, it depends how we want to convey the message. If we define cases with zero minimum availability or cases with positive/general minimum availability. 

We mention `minimum availability` and jump to explaining zero minimum availability. I think it might be useful to at least explain the term first, mainly for Rolling strategy (eg. just by mentioning `.spec.replicas - maxUnavailable` or `.spec.replicas - .spec.strategy.rollingUpdate.maxUnavailable`). Then we could define what zero minimum availability can happen.